datasets/makhzan/makhzan.py

```diff
@@ -0,0 +1,120 @@
+import xml.etree.ElementTree as ET
+from pathlib import Path
+
+import datasets
+
+
+_CITATION = """\
+@InProceedings{huggingface:dataset,
+title = {A great new dataset},
+authors={huggingface, Inc.
+},
+year={2020}
+}
+"""
+
+_DESCRIPTION = """\
+An Urdu text corpus for machine learning, natural language processing and linguistic analysis.
+"""
+
+_HOMEPAGE = "https://matnsaz.net/en/makhzan"
+
+_LICENSE = "All files in the /text directory are covered under standard copyright. Each piece of text has been included in this repository with explicity permission of respective copyright holders, who are identified in the <meta> tag for each file. You are free to use this text for analysis, research and development, but you are not allowed to redistribute or republish this text. Some cases where a less restrictive license could apply to files in the /text directory are presented below. In some cases copyright free text has been digitally reproduced through the hard work of our collaborators. In such cases we have credited the appropriate people where possible in a notes field in the file's metadata, and we strongly encourage you to contact them before redistributing this text in any form. Where a separate license is provided along with the text, we have provided corresponding data in the publication field in a file's metadata."
+
+_DOWNLOAD_URL = "https://github.com/zeerakahmed/makhzan/archive/master.zip"
+
+
+class Makhzan(datasets.GeneratorBasedBuilder):
+    """Makhzan - An Urdu text corpus for machine learning, natural language processing and linguistic analysis."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    def _info(self):
+        features = datasets.Features(
+            {
+                "file_id": datasets.Value("string"),
+                "metadata": datasets.Value("string"),
+                "title": datasets.Value("string"),
+                "num-words": datasets.Value("int64"),
+                "contains-non-urdu-languages": datasets.Value("string"),
+                "document_body": datasets.Value("string")
+                # These are the features of your dataset like images, labels ...
+            }
+        )
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            supervised_keys=None,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_DOWNLOAD_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"data_dir": data_dir},
+            )
+        ]
+
+    def _generate_examples(self, data_dir):
+        """ Yields examples. """
+        data_dir_path = Path(data_dir)
+        data_dir_path = data_dir_path / "makhzan-master" / "text"
+        file_paths = data_dir_path.glob(r"*.xml")
```

Comment:
```suggestion
        file_paths = sorted(data_dir_path.glob(r"*.xml"))
```
we need to make the order deterministic here